### PR TITLE
test: pkg/id

### PR DESCRIPTION
**Pull request by yk-eukarya**

### CODE REVIEW

File: pkg/id/id_test.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+		})
+	}
+}
+
+func TestID_Copy(t *testing.T) {
+	id := New()
+
+	id2 := id.Copy()
+
+	assert.Equal(t, id.id, id2.id)
```
Comment (rot1024) [inline pkg/id/id_test.go:225]:
```suggestion
	assert.Equal(t, id.id, id2.id)
	assert.NotSame(t, id.id, id2.id)
```

`NotSame` is also helpful to test if they are not the same pointer.

File: pkg/id/id_test.go [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+func TestID_Equal(t *testing.T) {
+	id1 := New()
+	id2 := id1.Copy()
+
+	assert.True(t, id1.Equal(id2))
+	assert.False(t, id1.Equal(New()))
+}
+
+func TestID_IsEmpty(t *testing.T) {
+
```
Comment (rot1024) [inline pkg/id/id_test.go:276]:
```suggestion
```

File: pkg/id/id_test.go [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+func TestID_parseID(t *testing.T) {
+	_, err := parseID("")
+
+	assert.True(t, errors.As(ErrInvalidID, &err))
+
+	id, err := parseID("01f2r7kg1fvvffp0gmexgy5hxy")
+
+	assert.Nil(t, err)
+	assert.EqualValues(t, strings.ToLower(id.String()), "01f2r7kg1fvvffp0gmexgy5hxy")
+
```
Comment (rot1024) [inline pkg/id/id_test.go:340]:
```suggestion
```

File: pkg/id/id_test.tmpl [hunk 4]
```diff
@@ -0,0 +1,1006 @@
...
+}
+
+func Test{{$name}}ID_CopyRef(t *testing.T) {
+	id := New()
+	subId := {{$name}}IDFromRefID(&id)
+
+	subIdCopyRef := subId.CopyRef()
+
+	assert.Equal(t, *subId, *subIdCopyRef)
+	assert.False(t, subId == subIdCopyRef)
```
Comment (rot1024) [inline pkg/id/id_test.tmpl:215]:
```suggestion
	assert.NotSame(t, subId, subIdCopyRef)
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: pkg/id/id_test.go (hunk 1 above)
Comment (yk-eukarya) [reply pkg/id/id_test.go:225]: Thanks I was looking how to assert the pointers are equal or not

**Review by rot1024 — APPROVED** [review]

**Pull request merged**